gcsio/src/main/java/com/google/cloud/hadoop/gcsio/storageapioverride/StorageRequestFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.google.cloud.hadoop.gcsio.storageapioverride;
```

Comment:
It seems like all classes in the `com/google/cloud/hadoop/gcsio/storageapioverride/` package are quite small. I'm thinking that it maybe worth to move them all as static nested classes in the `StorageRequestFactory` class and make `StorageRequestFactory` package-private in the `com/google/cloud/hadoop/gcsio/` package - this will allow us to change it freely without worrying about breaking changes as we would do for public classes.